examples/tutorial_examples/add_custom_bundle.py

```diff
@@ -0,0 +1,222 @@
+
+pyAFQ is designed to be customizable and extensible. This example shows how you
+can customize it to define a new bundle based on a definition of waypoint and
+endpoint ROIs of your design.
+
+In this case, we add the optic radiations, based on work by Caffara et al. [1]_,
+[2]_. The optic radiations (OR) are the primary projection of visual information
+from the lateral geniculate nucleus of the thalamus to the primary visual
+cortex. Studying the optic radiations with dMRI provides a linkage between white
+matter tissue properties, visual perception and behavior, and physiological
+responses of the visual cortex to visual stimulation.
+
+We start by importing some of the components that we need for this example and
+fixing the random seed for reproducibility
+
+"""
+
+import os.path as op
+import plotly
+import numpy as np
+import shutil
+
+from AFQ.api.group import GroupAFQ
+import AFQ.api.bundle_dict as abd
+import AFQ.data.fetch as afd
+from AFQ.definitions.image import ImageFile, RoiImage
+import AFQ.utils.streamlines as aus
+import wget
+import os
+np.random.seed(1234)
+
+
+#############################################################################
+# Get dMRI data
+# ---------------
+# We will analyze eight subject from the Healthy Brain Network Processed Open
+# Diffusion Derivatives dataset (HBN-POD2) [3]_, [4]_. We'll use a fetcher to
+# get preprocessed dMRI data for eight of the >2,000 subjects in that study. The
+# data gets organized into a BIDS-compatible format in the `~/AFQ_data/HBN`
+# folder. The fether returns this directory as study_dir:
+
+_, study_dir = afd.fetch_hbn_preproc([
+    'NDARAA948VFH', 'NDARXT727MD7', 'NDARXT822TAA', 'NDARXU376FDN', 
+    'NDARXU437UFZ', 'NDARXU679ZE8', 'NDARXU883NMY', 'NDARXV094ZHH'])
+
+
+
+roi_urls = ['https://github.com/yeatmanlab/AFQ/raw/c762ca4c393f2105d4f444c44d9e4b4702f0a646/SLF123/ROIs/MFgL.nii.gz',
+            'https://github.com/yeatmanlab/AFQ/raw/c762ca4c393f2105d4f444c44d9e4b4702f0a646/SLF123/ROIs/MFgR.nii.gz',
+            'https://github.com/yeatmanlab/AFQ/raw/c762ca4c393f2105d4f444c44d9e4b4702f0a646/SLF123/ROIs/PaL.nii.gz',
+            'https://github.com/yeatmanlab/AFQ/raw/c762ca4c393f2105d4f444c44d9e4b4702f0a646/SLF123/ROIs/PaR.nii.gz',
+            'https://github.com/yeatmanlab/AFQ/raw/c762ca4c393f2105d4f444c44d9e4b4702f0a646/SLF123/ROIs/PrgL.nii.gz',
+            'https://github.com/yeatmanlab/AFQ/raw/c762ca4c393f2105d4f444c44d9e4b4702f0a646/SLF123/ROIs/PrgR.nii.gz',
+            'https://github.com/yeatmanlab/AFQ/raw/c762ca4c393f2105d4f444c44d9e4b4702f0a646/SLF123/ROIs/SFgL.nii.gz',
+            'https://github.com/yeatmanlab/AFQ/raw/c762ca4c393f2105d4f444c44d9e4b4702f0a646/SLF123/ROIs/SFgR.nii.gz',
+            'https://github.com/yeatmanlab/AFQ/raw/c762ca4c393f2105d4f444c44d9e4b4702f0a646/SLF123/ROIs/SLFt_roi2_L.nii.gz',
+            'https://github.com/yeatmanlab/AFQ/raw/c762ca4c393f2105d4f444c44d9e4b4702f0a646/SLF123/ROIs/SLFt_roi2_R.nii.gz']
```

Comment:
This is actually completely fine by me. The ROIs can come from anywhere. The only thing that this doesn't do that the fetcher automatically does for us is validation of the copied files, so if a file is corrupted in the download process, the user will not know about this until somewhat further down the line, when the program produces some unexpected results. Also, the last ROI here is duplication of an ROI that is already in the standard set of ROIs, so we could also get it from there.